gateway/modules/auth/errors.go

```diff
@@ -10,3 +15,16 @@ var ErrIncorrectRuleFieldType = errors.New("auth: Incorrect rule field type")
 
 // ErrIncorrectMatch is thrown when the field type of a rule is of incorrect type
 var ErrIncorrectMatch = errors.New("auth: The two fields do not match")
+
+//Formaterror check whether error is provided in config.Rule
+func formatError(rule *config.Rule, err error) error {
+	if rule.Name == "" {
```

Comment:
The first thing you should do is check if err you receive as a parameter is nil. Simply return nil if thats the case